marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -760,6 +776,17 @@ private void runStatusListener(long totalRunBatches, double batchCount, AtomicLo
             });
         }
     }
+    //percentComplete for csv files is (csvFilesProcessed/ urisCount) * 100.0
+    //The number of csv files would probably be less than that of regular files, so the step status listeners are updated more frequently
+    private void runCsvStatusListener(double urisCount, AtomicLong successfulEvents, AtomicLong failedEvents) {
```

Comment:
I think it's worth making this protected so that some unit tests can be written against it. You can call onStatusChanged on WriteStepRunner and pass in a test implementation of StepStatusListener. That implementation can save the args that are passed to it by this code, and your test can then verify that it received the expected args.